### DISCUSSION TITLE
Visualization of IFCSpace

### DISCUSSION THREAD

**Discussion by chetan-set** [IfcOpenShell/IfcOpenShell#4272] [General]
Hi everyone,

I'm facing an issue and would love your input. 

My question is this:

How is an IFCSpace meant to be rendered?

I have a model with lots of IFCSpace and BlenderBIM renders it like a glass-like texture. However if I use ifcConvert and convert the model into a GLB file, it is rendered as empty space.

I was wondering what would you consider as the correct approach here?

This is the blenderBIM output:

<img width="718" alt="Screenshot 2024-01-31 at 2 13 12 PM" src="https://github.com/IfcOpenShell/IfcOpenShell/assets/89379991/6cfb4ce2-df91-46d9-a70f-e34bf6ec17b0">

This is the output on ThreeJS after converting the ifc file to glb using IfcConvert

<img width="718" alt="Screenshot 2024-01-31 at 2 20 43 PM" src="https://github.com/IfcOpenShell/IfcOpenShell/assets/89379991/43b7e44d-38cd-445d-bf00-6f2c19cbe400">

Thank you

**Comment by aothms** [comment]
IfcConvert by default excludes spaces (and opening elements) in its output. You can run `IfcConvert clinic_a.ifc clinic_a.glb --exclude entities IfcOpeningElement` to override the default exclude filter. This is somewhat confusing behaviour.

**Comment by Moult** [comment]
IfcSpace is visualised in Blender using whatever style is defined in the IFC. If your IFC defines it as a transparent glass like texture, then that's what it'll display as.

Many viewers (or IfcConvert) tend to exclude the space by default (and some never allow you to view them). It's not right or wrong, just a default :)

**Reply by chetan-set** [reply]
Thanks so much guys! I just wanted to reconcile the two behaviours. 

With the flag, this is the output: 

<img width="718" alt="Screenshot 2024-01-31 at 5 45 30 PM" src="https://github.com/IfcOpenShell/IfcOpenShell/assets/89379991/b79317c1-a510-411e-9706-b781e5caaa80">
